### PR TITLE
Connect to API

### PR DESCRIPTION
**Pull request by getssh**
- Add axios to project
- Connected api with redux
- Used createAsyncThunk for redux async
- Handled GET, POST, and DELETE methods
- Check all linter errors

### CODE REVIEW

File: src/components/Book.js [hunk 1]
```diff
@@ -1,25 +1,51 @@
...
+import { getBooks } from '../redux/books/booksSlice';
 
 const Book = () => {
-  const books = useSelector((state) => state.books.books);
+  const {
+    books, isLoading, error, done,
+  } = useSelector((state) => state.books);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(getBooks());
```
Comment (iambenkis) [inline src/components/Book.js]: - [x] Good job adding loading the books from the API using the lines below, however when you add a book it should be rendered directly on the screen , It requires the user to reload the page, so he may see the added (or deleted ) book. The dependency you use does not allow you to get the real-time changes 👍   Kindly change the dependency from `[dispatch, done]` into `[books]`

```bash
  useEffect(() => {
    dispatch(getBooks());
  }, [books]);
```

Check [this link](https://blog.bitsrc.io/understanding-dependencies-in-useeffect-7afd4df37c96) for more about dependency 😉

**Review by iambenkis — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi @getssh   
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 

- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @iambenkis  when you comment, so I can receive the notification or use Slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/components/Book.js (hunk 1 above)
Comment (getssh) [reply src/components/Book.js]: Thank you for the review. I have implemented the recommended change

**Review by GedeonTS — APPROVED** [review]
# STATUS: APPROVED :tada: :clap: :champagne: 
Hi @getssh ,

You are amazing :smiley: :clap: !  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

 

## Highlights :clap: :tophat: 
- No linter errors :+1: 
- Good use of Gitflow :+1: 
- The component is rerendering perfectly :+1: 
## Optional suggestions :small_airplane: 

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**